### PR TITLE
:book: Fix typo in docs for TopologyDryRunAnnotation

### PR DESCRIPTION
**Pull request by guettli**
**What this PR does / why we need it**:

In the docstring of `TopologyDryRunAnnotation` the word "deny" was used, although it should be "skip".

/area documentation

### CODE REVIEW

File: api/v1beta1/common_types.go [hunk 1]
```diff
@@ -160,7 +160,7 @@ const (
 	// only during a server side dry run apply operation. It is used for validating
 	// update webhooks for objects which get updated by template rotation (e.g. InfrastructureMachineTemplate).
 	// When the annotation is set and the admission request is a dry run, the webhook should
-	// deny validation due to immutability. By that the request will succeed (without
+	// skip validation due to immutability. By that the request will succeed (without
 	// any changes to the actual object because it is a dry run) and the topology controller
 	// will receive the resulting object.
```
Comment (schrej) [inline api/v1beta1/common_types.go:165]: Maybe rephrase it even more, so it's clearer?

> TopologyDryRunAnnotation is an annotation that gets set on objects by the topology controller when performing a dry-run server side apply in order to apply defaulting to resources. It instructs the validating web hook to skip validation of immutable fields, which allows the request to succeed. The annotation must not be respected for regular (not dry-run) requests, so no invalid modifications to the resources are applied.
The topology controller uses this feature when processing immutable resources (usually templates) to decide whether a new revision of a immutable resource needs to be created.

**Comment by fabriziopandini** [comment]
/ok-to-test

it probably requires to run make generate-manifests

**Comment by enxebre** [comment]
Thanks @guettli are you still planning to address the suggestion here https://github.com/kubernetes-sigs/cluster-api/pull/10265#discussion_r1523024698?

**Comment by guettli** [comment]
@enxebre I think I am not familiar enough with it, to write a better comment. Feel free to adapt the comment/PR, so that it makes more sense.

**Comment by enxebre** [comment]
Ok, thanks! We can always follow up.
/lgtm
/approve
